pkg/integration/replication/suite.go

```diff
@@ -44,8 +44,9 @@ type baseReplicationTestSuite struct {
 	mu sync.Mutex
 
 	// server settings
-	master    TestServer
-	followers []TestServer
+	master           TestServer
+	followers        []TestServer
+	followersRunning []bool
```

Comment:
IMO, if we pop the follower from the list (treat it like a queue and remove the last item), we won't need this array. Else it is a lot of state to maintain. WDYT?

Reply:
```
// get last item
f:=followers[len(followers)-1]
// pop
followers = followers[:len(followers]
```

something similar

Reply:
I wanted to keep follower on that list to allow starting it with the same data folder (and in some tests I want to explicitly start again the same instance).

The `followersRunning` added here is only a workaround for the TearDown method which I didn't notice earlier - because we don't allow double shutdowns.

Reply:
ack, makes sense to know the state